integration-tests/test/test_wallets.py

```diff
@@ -73,6 +81,36 @@ def get_vault_balance(context: TestingContext, node: Node, rev_addr: str, privat
     check_balance_match = wait_for_log_match_result(context, node, check_balance_pattern)
     return (blockHash, int(check_balance_match.group("balance")))
 
+
+def test_transfer_from_pos_vault(command_line_options: CommandLineOptions, docker_client: DockerClient, random_generator: Random) -> None:
+    pos_vault_private_key = PrivateKey.generate()
+    genesis_vault = {
+        pos_vault_private_key: 50000000,
+        BOB_KEY: 0
+    }
+
+    pos_vault_pub_key = pos_vault_private_key.get_public_key().to_hex()
+    with testing_context(command_line_options, random_generator, docker_client, wallets_dict=genesis_vault) as context, \
+            started_bootstrap_with_network(context=context, pos_vault_pub_key=pos_vault_pub_key) as bootstrap:
+        wait_for_approved_block_received_handler_state(context, bootstrap)
+        transfer_amount = 100000
+        pos_vault_rev_address = pos_vault_private_key.get_public_key().get_rev_address()
+
+        bob_rev_address = BOB_KEY.get_public_key().get_rev_address()
+        _, pos_vault_balance = get_vault_balance(context, bootstrap, pos_vault_rev_address, pos_vault_private_key, 1000000, 1)
+        _, bob_balance = get_vault_balance(context, bootstrap, bob_rev_address, pos_vault_private_key, 1000000, 1)
+        assert pos_vault_balance == 50000000 - 1000000
+        assert bob_balance == 0
+
+        log_marker = random_string(context, 10)
+        transfer_funds_result_pattern = re.compile('"{} (Successfully|Failing) reason: (?P<reason>[a-zA-Z0-9 ]*)"'.format(log_marker))
+        deploy_transfer_from_pos_vault(log_marker, bootstrap, bob_rev_address, transfer_amount, pos_vault_private_key, 1000000, 1)
```

Comment:
Could you please add one more step in the end and test that transfer from not `pos_vault_private_key` would fail?